RecoLocalCalo/HcalRecProducers/src/HcalReconstructor.cc

```diff
@@ -0,0 +1,118 @@
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/HcalRecHit/interface/HcalRecHitCollections.h"
+#include "HeterogeneousCore/SonicTriton/interface/TritonClient.h"
+#include "HeterogeneousCore/SonicCore/interface/SonicEDProducer.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "DataFormats/HcalRecHit/interface/HBHERecHit.h"
+
+class HcalReconstructor : public SonicEDProducer<TritonClient>
+{
+public:
+    explicit HcalReconstructor(edm::ParameterSet const& cfg) : 
+        SonicEDProducer<TritonClient>(cfg),
+        fChannelInfoName_(cfg.getParameter<edm::InputTag>("ChannelInfoName")),
+        fTokChannelInfo_(this->consumes<HBHEChannelInfoCollection>(fChannelInfoName_))
+     
+    {
+        this->produces<HBHERecHitCollection>();
+        this->setDebugName("HcalReconstructor");
+    }
+
+    void acquire(edm::Event const& iEvent, edm::EventSetup const& iSetup, Input& iInput) override {
+
+	const HBHEChannelInfoCollection *channelInfo   = 0;
+	edm::Handle<HBHEChannelInfoCollection> hChannelInfo;
+	iEvent.getByToken(fTokChannelInfo_, hChannelInfo); 
+	channelInfo = hChannelInfo.product();
+
+	auto& input1 = iInput.begin()->second;
+	auto data1 = std::make_shared<TritonInput<float>>();
+	data1->reserve(std::distance(channelInfo->begin(), channelInfo->end()));
+
+	hcalIds_.clear(); 
+	
+	for(HBHEChannelInfoCollection::const_iterator itC = channelInfo->begin(); itC != channelInfo->end(); itC++){
+
+	    std::vector<float> input;
+	    const HBHEChannelInfo& pChannel(*itC);
+  	    const HcalDetId        pDetId = pChannel.id();
+ 	    hcalIds_.push_back(pDetId);    
+
+	    //FACILE uses iphi as a continuous variable
+	    input.push_back((float)pDetId.iphi());
+            input.push_back((float)pChannel.tsGain(0.));
```

Comment:
the dev version uses `respcorrgain`: https://github.com/hls-fpga-machine-learning/SonicCMS/blob/97ae4c4fa5a791d501f76e1bce7072cca8d8a791/TensorRT/plugins/HcalPhase1Reconstructor_FACILE.cc#L197
did you check if `tsGain` is the same value?

Reply:
`pChannel.tsGain(0.)` is the gain value that Facile is trained on